src/edu/washington/escience/myria/operator/SampledDbInsertTemp.java

```diff
@@ -0,0 +1,156 @@
+/**
+ * 
+ */
+package edu.washington.escience.myria.operator;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+import edu.washington.escience.myria.DbException;
+import edu.washington.escience.myria.RelationKey;
+import edu.washington.escience.myria.Schema;
+import edu.washington.escience.myria.Type;
+import edu.washington.escience.myria.accessmethod.ConnectionInfo;
+import edu.washington.escience.myria.column.Column;
+import edu.washington.escience.myria.column.builder.IntColumnBuilder;
+import edu.washington.escience.myria.parallel.RelationWriteMetadata;
+import edu.washington.escience.myria.storage.MutableTupleBuffer;
+import edu.washington.escience.myria.storage.TupleBatch;
+
+/**
+ * Samples the stream into a temp relation.
+ */
+public class SampledDbInsertTemp extends DbInsertTemp implements DbWriter {
```

Comment:
Nitpick: the interface `DbWriter` is already implemented by the grandparent (`AbstractDBInsert`), so is redundant here.